### ISSUE TITLE
[YSQL] Create-Table-As-Select needs option to specify split-into-n-tablets. 

### ISSUE THREAD

**Issue by pdvmoto** [yugabyte/yugabyte-db#21061]
### Description

Request: include the option to specify "split into n tablets" on Create-Table-As-Select (CTAS)

Ive created several tables with CTAS, discovered I could not specify nr of tablets.
I would like to determine the exact nr of tablets for each table (some are small, some are large and need more tablets). 
There is no option in the CTAS to specify nr of tables (yet).
https://docs.yugabyte.com/preview/api/ysql/the-sql-language/statements/ddl_create_table_as/

### Issue Type

kind/enhancement

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

**Comment by FranckPachot** [comment]
Yes, I agree this could be helpful, but this also means that we need to be able to define the primary key in a CTAS, which doesn't exists in PostgreSQL
```
yugabyte=# drop table if exists demo; create table demo (id, primary key(id)) as select 1;
DROP TABLE
ERROR:  42601: syntax error at or near "primary"
LINE 1: create table demo (id, primary key(id)) as select 1;
                               ^
LOCATION:  scanner_yyerror, scan.l:1133
```
